### PR TITLE
[Merged by Bors] - feat: singletons are closed in T1 spaces

### PR DESCRIPTION
**Pull request by scholzhannah**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
Thanks for your contribution! This looks good to me, let's have a maintainer take a look!
maintainer merge

File: Mathlib/Topology/Separation.lean [hunk 1]
```diff
@@ -1068,6 +1068,17 @@ theorem SeparationQuotient.t1Space_iff : T1Space (SeparationQuotient X) ↔ R0Sp
     erw [mk_eq_mk, inseparable_iff_specializes_and]
     exact ⟨xspecy, yspecx⟩
 
+lemma Set.Subsingleton.isClosed [T1Space X] {A : Set X} (h : A.Subsingleton) : IsClosed A := by
+  rcases h.eq_empty_or_singleton with rfl | ⟨x, rfl⟩
+  · exact isClosed_empty
+  · exact isClosed_singleton
+
+lemma IsClosed.inter_singleton [T1Space X] {A : Set X} {a : X} : IsClosed (A ∩ {a}) :=
```
Comment (grunweg) [inline Mathlib/Topology/Separation.lean:1076]: Upon second thought, I think `isClosed_inter_singleton` would be a better name.
*If* this lemma took a hypothesis `IsClosed A`, `IsClosed.inter_singleton` would be a great name and allow access to dot notation. That is not the case here. Same comment applies below.

**Review by grunweg — COMMENTED** [review]
Actually, I have one comment about naming.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**